common/src/test/scala/explore/utils/StreamSpec.scala

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.utils
+
+import cats.effect
+import cats.effect.IO
+import cats.effect.testkit.TestControl
+import fs2.Stream
+import munit.Location
+
+import scala.concurrent.duration.*
+
+class StreamSpec extends munit.CatsEffectSuite:
```

Comment:
Sorry if I missed it, is there a test here that asserts the behavioral difference from `groupWithin`? Something to guard against future regression in case somebody reading the code thinks they can simplify it.

Reply:
I _think_ with groupWithin at least one of the tests would fail (it would emit more elements). But I'll confirm it tomorrow

Reply:
Changing the implementation to groupWithin does make at least 1 test fail, but I added another just to make it explicit